src/parser/expression/star_expression.cpp

```diff
@@ -167,4 +159,45 @@ qualified_column_set_t StarExpression::SerializedQualifiedExcludeList() const {
 	return result;
 }
 
+void StarExpression::Serialize(Serializer &serializer) const {
+	ParsedExpression::Serialize(serializer);
+	serializer.WritePropertyWithDefault<string>(200, "relation_name", relation_name);
+	serializer.WriteProperty<case_insensitive_set_t>(201, "exclude_list", SerializedExcludeList());
+	serializer.WritePropertyWithDefault<case_insensitive_map_t<unique_ptr<ParsedExpression>>>(202, "replace_list",
+	                                                                                          replace_list);
+	serializer.WritePropertyWithDefault<bool>(203, "columns", columns);
+	serializer.WritePropertyWithDefault<unique_ptr<ParsedExpression>>(204, "expr", expr);
+	serializer.WritePropertyWithDefault<qualified_column_set_t>(
+	    206, "qualified_exclude_list", SerializedQualifiedExcludeList(), qualified_column_set_t());
+	serializer.WritePropertyWithDefault<qualified_column_map_t<string>>(207, "rename_list", rename_list,
+	                                                                    qualified_column_map_t<string>());
+}
+
+unique_ptr<ParsedExpression> StarExpression::Deserialize(Deserializer &deserializer) {
+	auto relation_name = deserializer.ReadPropertyWithDefault<string>(200, "relation_name");
+	auto exclude_list = deserializer.ReadProperty<case_insensitive_set_t>(201, "exclude_list");
+	auto replace_list =
+	    deserializer.ReadPropertyWithDefault<case_insensitive_map_t<unique_ptr<ParsedExpression>>>(202, "replace_list");
+	auto columns = deserializer.ReadPropertyWithDefault<bool>(203, "columns");
+	auto expr = deserializer.ReadPropertyWithDefault<unique_ptr<ParsedExpression>>(204, "expr");
+	auto unpacked = deserializer.ReadPropertyWithExplicitDefault<bool>(205, "unpacked", false);
+	auto qualified_exclude_list = deserializer.ReadPropertyWithExplicitDefault<qualified_column_set_t>(
+	    206, "qualified_exclude_list", qualified_column_set_t());
+	auto result = duckdb::unique_ptr<StarExpression>(new StarExpression(exclude_list, qualified_exclude_list));
+	result->relation_name = std::move(relation_name);
+	result->replace_list = std::move(replace_list);
+	result->columns = columns;
+	result->expr = std::move(expr);
+	deserializer.ReadPropertyWithExplicitDefault<qualified_column_map_t<string>>(
+	    207, "rename_list", result->rename_list, qualified_column_map_t<string>());
+	if (unpacked) {
```

Comment:
Maybe we can adjust the generation script to invoke a callback to construct the result instead of having a constructor, then we can avoid having to have a custom implementation here

Reply:
e.g. `"constructor_method": "DeserializeStarExpression"` which then generates `StarExpression::DeserializeStarExpression(...)`

Reply:
Implemented this, much cleaner for sure